lib/plug/static.ex

```diff
@@ -182,6 +186,100 @@ defmodule Plug.Static do
     h in only or match?({0, _}, prefix != [] and :binary.match(h, prefix))
   end
 
+  defp serve_range(conn, path, segments, [], %{gzip?: gzip?, brotli?: brotli?} = options) do
+    encoding = file_encoding(conn, path, gzip?, brotli?)
+    serve_static(encoding, segments, options)
+  end
+  defp serve_range(conn, path, segments, [range], options) do
+    encoding = file_encoding(conn, path, false, false)
+    serve_range(encoding, range, segments, options)
+  end
+  defp serve_range(_conn, _path, _segments, _range, _options) do
+    raise InvalidRangeError
+  end
+
+  @byte_range_pattern ~r/^\s*bytes=([0-9]+)?-([0-9]+)?\s*$/
```

Comment:
Please parse the bytes accordingly instead of relying on regexes. See the functions in Plug.Conn.Utils that can help you parse params. If necessary, add new functions there.